### ISSUE TITLE
LTSB/LTSC support

### ISSUE THREAD

**Issue by dimqua** [pbatard/Fido#8]
Is it possible to add a support for those versions to this script?

**Comment by pbatard** [comment]
If Microsoft does make them available publicly (which AFAIK I don't believe they do) and you can find their `Product Edition ID` (the last number like `1060` or `1056` you can see [here](https://github.com/pbatard/Fido/blob/master/Fido.ps1#L74-L188)) sure.

Otherwise no. Remember, this script only allows access to what ISOs Microsoft allows users to __publicly__ and __freely__ download, and it seems that LTSC are not in the group.

That's also the reason why we don't provide Windows 7 ISOs, as Microsoft doesn't allow public access to them unless you first provide a valid license-key...

**Comment by dimqua** [comment]
It's available semi-publicly, all you need to do to get the link: go to [here](https://www.microsoft.com/en-us/evalcenter/evaluate-windows-10-enterprise), select LTSC, enter any (fake) data, select preferred language and architecture.

**Comment by pbatard** [comment]
I had a look but it seems to me like it's going to be a PITA to add and maintain compared to the number of people whom it might benefit (from what I gather you need a special license key for LTSC, which is reserved for enterprise customer, whereas this script is aimed at the general public since it is designed to be used with Rufus).

Also the script mostly exists because Microsoft actively prevents Windows 10 users from downloading Windows 10 retail ISOs (i.e. if you go to https://www.microsoft.com/software-download/windows10ISO Microsoft will direct you to away from the page where you can download the ISOs).

On the other hand, https://www.microsoft.com/en-us/evalcenter/evaluate-windows-10-enterprise does let you access the ISOs regardless of the OS you use of where you come from, so I will assert that the minority of people who really want to download the LTSC version will also know about that page and be able to download those ISOs independently.

In other words, the cost/benefit of adding LTSC support to the script is simply too high (bearing in mind that I am trying to keep the script relatively small since it is being downloaded on the fly by Rufus), so, all things considered, I regret to inform you that __I__ am not planning to add it.

However, if someone provides a patch, and if that patch doesn't make the script size explode, I may consider adding it...

I will therefore close this issue. But thanks for posting the link, as it means that users who really want LTSC ISOs will stll be able to download them.

**Comment by pbatard** [comment]
PS: I'll see if I can add a note in the readme about LTSC ISO downloads and pointing people who are interested in those to the LTSC download page.

**Comment by dimqua** [comment]
Isn't allowed to just hardcore those links? Like:
* en-US x86_64: https://software-download.microsoft.com/download/pr/17763.107.101029-1455.rs5_release_svc_refresh_CLIENT_LTSC_EVAL_x64FRE_en-us.iso
* en-GB x86_64: https://software-download.microsoft.com/download/pr/17763.107.101029-1455.rs5_release_svc_refresh_CLIENT_LTSC_EVAL_x64FRE_en-gb.iso
* etc.

**Comment by pbatard** [comment]
Except those are just for English. I'd need all the languages, so that's about 35 x 2. And I'd need to make sure it integrates in the current process which is designed to query the MS APIs. Plus what happens when MS starts to create releases other than 17763.107? Are there other releases?

I see a script size increase of at least 15% to provide LTSC support, and that's not mentioning the maintenance part (making sure that these scripts and links stay valid, whereas, for the regular retail ISOs we just get the links from the MS APIs). Again, as far as I'm concerned, the cost/benefit ratio is simply too high for ISOs that are aimed at Enterprise customers and which almost none of the people who use Rufus would be able to install legally.
